### PR TITLE
Set a limit the area size the users can create

### PR DESCRIPTION
**Pull request by jonasblanc**
If the area size defined by the three control vertices exceed 10'000 m^2 the user can't preview the mission nor save it.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt [hunk 1]
```diff
@@ -94,6 +95,9 @@ class ItineraryCreateActivity : BaseMapActivity(), OnMapReadyCallback,
         const val FLIGHTHEIGHT_INTENT_PATH = "ICA_flightHeight"
         const val DEFAULT_FLIGHTHEIGHT = 50.0
         val DEFAULT_STRATEGY = Strategy.SINGLE_PASS
+
+        // Maximum area size in m2
+        const val MAXIMUM_AREA_SIZE = 100000.0
```
Comment (Squalene) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt:100]: A bit oversized maybe 10 000m^2 would be better

Comment (jonasblanc) [reply app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt:100]: Yes your right, I'm bad at maths, fortunately we don't get tested on math at EPFL

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/map/area/ParallelogramAreaBuilderTest.kt [hunk 2]
```diff
@@ -56,4 +57,25 @@ class ParallelAreaBuilderTest {
...
         assertThat(area.vertices.size, equalTo(0))
     }
+
+    @Test
+    fun getAreaSizeIsCorrect() {
+        val area = ParallelogramBuilder()
+
+        val a = LatLng(Random.nextDouble(), Random.nextDouble())
+        val b = LatLng(Random.nextDouble(), Random.nextDouble())
+        val c = LatLng(Random.nextDouble(), Random.nextDouble())
```
Comment (Jorque) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/map/area/ParallelogramAreaBuilderTest.kt:67]: Random.nextDouble() gives values between 0 and 1 and I understand that it is to not create a too big area but it causes to only test within this small range. To diversify the testing, we could generate a random latitude and longitude from which we will create the area. Maybe looking at my code suggestion gives you an idea of what I'm thinking.
```suggestion
        val initialLat = Random.nextDouble(-90.0, 89.0)
        val initialLng = Random.nextDouble(-180.0, 179.0)
        val a = LatLng(initialLat + Random.nextDouble(), initialLng + Random.nextDouble())
        val b = LatLng(initialLat + Random.nextDouble(), initialLng + Random.nextDouble())
        val c = LatLng(initialLat + Random.nextDouble(), initialLng + Random.nextDouble())
```

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt [hunk 3]
```diff
@@ -316,8 +328,22 @@ class ItineraryCreateActivity : BaseMapActivity(), OnMapReadyCallback,
...
+
+    /**
+     * Test if the size of the area is with the accepted size
+     */
+    private fun isSizeWithinLimit(): Boolean {
+        return if (areaBuilder.isComplete()) {
+            areaBuilder.getAreaSize() < MAXIMUM_AREA_SIZE
+        } else {
+            false
+        }
```
Comment (Jorque) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt:343]: You could have wrote it using && due to the short-circuit evaluation
```suggestion
        return areaBuilder.isComplete() && areaBuilder.getAreaSize() < MAXIMUM_AREA_SIZE
```

**Review by Jorque — APPROVED** [review]

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/map/area/ParallelogramAreaBuilderTest.kt (hunk 2 above)
Comment (jonasblanc) [reply app/src/androidTest/java/ch/epfl/sdp/drone3d/map/area/ParallelogramAreaBuilderTest.kt]: You're right thanks

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt (hunk 3 above)
Comment (jonasblanc) [reply app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt]: Nice one !

**Review by GabrielFleischer — APPROVED** [review]
Simple yet without issues.

Good job

**Pull request merged**